Domains: Update domain suggestions cards layout

leonardost opened the pull request:
#### Changes proposed in this Pull Request

This PR updates the layout of domain suggestions cards in the domain registration search page. This corresponds to task 8 of pcYYhz-e4-p2, which is part of a redesign of domains pages.

The main updates in this PR are:

- Featured domain suggestions are always shown in a full-width row - previously, they could be shown as two side-by-side cards if the searched domain name was short enough
- The colored bars indicating the quality of the suggestion ("Best match" or "Best alternative") were removed. They were replaced by badges with "Recommended" and "Best alternative" texts
- Both featured domain suggestions have the "Selected" button colored as primary now
- The "Sale" badge was reskinned to have the same format as the "Recommended" and "Best alternative" ones

This is what the domain suggestions page looked like before:

<img width="1055" alt="Screen Shot 2021-08-02 at 20 00 14" src="https://user-images.githubusercontent.com/5324818/127934732-3c1cf44d-d9df-4259-b5f3-e6dad631a315.png">

And what it looks like with this update:

<img width="1055" alt="Screen Shot 2021-08-11 at 12 35 25" src="https://user-images.githubusercontent.com/5324818/129059585-e55515d5-3ced-4bd0-b089-2e45e5c04ff6.png">

Mobile view before:

<img width="417" alt="Screen Shot 2021-08-02 at 20 02 44" src="https://user-images.githubusercontent.com/5324818/127934845-5bf52beb-88b3-494f-a216-8a2e1fa37725.png">

Mobile view after:

<img width="429" alt="Screen Shot 2021-08-11 at 12 36 42" src="https://user-images.githubusercontent.com/5324818/129059616-983ab796-92ed-4779-9ce8-b6414f6d72ac.png">

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Use the live Calypso link and navigate to the domain name search page (Upgrades > Domains > Add a domain > To this site). Do a few domain searches with different names and TLDs and ensure everything looks alright. Please test both desktop and mobile views to ensure both are OK.

It would also be great if you could take a look at the signup flow (`<calypso live URL>/start`) and check that the domain search there looks OK too, because it uses the same components that were updated here. 🙇‍♂️ It was a little tricky to not change things in that page.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

leonardost committed:
Change domain search page title to "Search for a domain"

leonardost committed:
Make suggested domain cards always occupy full width

leonardost committed:
Move "Select" buttons to the right side of the featured suggestion card

leonardost committed:
Color the secondary featured suggestion "Select" button in the primary color

leonardost committed:
Align badge and "Select" button correctly in featured domain suggestion card

leonardost committed:
Justify common domain suggestion name and price correctly

leonardost committed:
Show checkmark items only in domains page and not in signup

leonardost committed:
Move price to the right in the featured domain suggestion card

leonardost committed:
Restyle pricing text

leonardost committed:
Align "Select" button and price correctly in featured domain suggestion

leonardost committed:
Fix price text layout in signup flow

leonardost committed:
Fix signup domain suggestions card height

leonardost committed:
Remove ProgressBar component and show price correctly

leonardost committed:
Change "Best match" badge for "Recommended"

leonardost committed:
Increase margins around domain name

leonardost committed:
Fix position of "Select" button

leonardost committed:
Style TLDs as bold in common domain suggestions

leonardost committed:
Corect badge margins in mobile view

leonardost committed:
Correct price position in mobile view

leonardost committed:
Style "Sale" badge to be like "Recommended" and "Best alternative" badges

leonardost committed:
Fix paddings of featured domain suggestions in signup

leonardost committed:
Fix E2E tests

poligilad-auto reviewed (changes requested):
Nice work @leonardost! Excuse me for the long list but these are a lot of changes :) 

General comments:

1. Make the additional domains (below the two suggested ones) vertically aligned. Seems like this can be done by removing `margin-bottom: 8px;` from `.domain-registration-suggestion__title-wrapper`.
2. Make `.domain-registration-suggestion__domain-title-name`  `var(--studio-gray-60);`.
3. Make `.domain-registration-suggestion__domain-title-tld`  `var(--studio-gray-90);`.
4. Make `.domain-registration-suggestion__domain-title`  `$font-body`.
5. All the _select buttons_ should be 40px high, not sure why they are all 31px now.
6. We have this strange _loading_ state for all the select buttons when clicking on one. Can we get rid of the loading state altogether?
7. _Close match etc._ should be `$font-body-small`.

**Align price display with onboarding:**
When the first year is free it should look like this. Wanted to double check this with you @kelliepeterson 
![image](https://user-images.githubusercontent.com/83961704/128172095-d98e19e2-6442-42a5-902d-a99b09ba113b.png)

1. Make .`domain-product-price__free-text` font-size `$font-body-small`, font-weight `500`., color: `var( --color-success-60 );`.
2. Make `.domain-product-price__price` `$font-body-small` and with a strikethrough.

When the domain is paid for it should look like this:
![image](https://user-images.githubusercontent.com/83961704/128173592-64e6a21a-d8c2-4720-a67b-c003c1e28fe3.png)

1. Make the domain price `font-weight 500`.
2. Make the /year `$font-body-small`.

Domains on sale:
Should look like this. Wanted to double check this with you @kelliepeterson.
![image](https://user-images.githubusercontent.com/83961704/128173684-d9a1b47f-1afd-41b0-8f89-cab005cdebf6.png)

1. The badge: make the background color  `var(--color-warning-10);`.
2. Put the _sale badge_ after the _best alternative badge_.
3. Make `.domain-product-price__sale-price` font-weight 600 & color `var( --studio-orange-50 );`
4. Change the copy.

For mobile: 
1. Just noticed when I see _best alternative_ & the _sale_ badge that it breaks the layout.

I'll go over it again afterwards and also checkout mobile more in depth.

Thank you!

delputnam reviewed (approved):
From a code perspective, this LGTM. Nice job on keeping changes here separate from the reskinned /start flow. 😅

leonardost committed:
Adjust CSS styles according to design review

leonardost committed:
Style pricing for domain registrations correctly

leonardost committed:
Fix style change in signup flow

leonardost committed:
Correct domain registration price styling

leonardost committed:
Contain all badges inside the same component

leonardost committed:
Fix badges' CSS styles

leonardost committed:
Align info bubble with domain name

leonardost committed:
Fix styling of premium badge in signup flow

leonardost committed:
Update copy of pricing for domains with TLDs on sale

leonardost commented:
Hi @poligilad-auto, thank you for the thorough review! I think I addressed all your observations 😄 

I have two questions:

- I've changed the button behavior to only show the "busy" state on the button we click on, all the others are disabled. It looks like this:

https://user-images.githubusercontent.com/5324818/128427084-779a6aeb-362e-48fd-a8cd-9a95fe55b154.mp4

Do you think that looks fine or is it still weird?

- I'm not sure about the copy for domains on sale. Do you think we should remove the "Renews at" text? Will users understand the value below is the renewal price for the first year afterwards? What do you think @kelliepeterson?

<img width="1067" alt="Screen Shot 2021-08-05 at 19 06 41" src="https://user-images.githubusercontent.com/5324818/128427475-7ec0d089-30ef-4e95-84cb-50d2d22588d3.png">

leonardost committed:
Update domain registration button behavior - only clicked button becomes busy

leonardost committed:
Rename CSS class `domain-registration-suggestion__progress-bar` to `domain-registration-suggestion__badges`

poligilad-auto commented:
Thanks for all the changes @leonardost, looking much better.

**Some small clarifications:**
1. When the domain is free: "Free for the first year" should be `$font-body-small`. The price below should be font-weight 400.
2. For the sale price: The sale price should stay `$font-body`, but the text "for the first year" should be `$font-body-small`. The text below should all be`$font-body-small` and with a strikethrough. That aligns with the way we display the price when the domain is free. The strikethrough price indicates the regular price and the sale price is above. I think that's clear. @kelliepeterson have any thoughts?
![image](https://user-images.githubusercontent.com/83961704/128682980-f3ca5f45-1715-48d8-bab4-fcbf461a9ef4.png)

> I've changed the button behavior to only show the "busy" state on the button we click on, all the others are disabled. It looks like this:

That's a setup in the right direction :) I'm not fan of this type of loading state, it's a bit outdated. 
For this I want to ask, do we actually need a loading state or could we just move on to the next step here? We barely use this in Dotcom, I only noticed it in the checkout.

Also, I noticed that we see the "In cart" button + the cart on top for one second before moving on, could we not show this and just move on to the next step?

**One other small thing:**
With the two "recommended" domains, when we have no explanations below the domain name isn't aligned to the center, cause there is some extra padding there.

leonardost committed:
Update style of domain prices

leonardost committed:
Fix featured domain name alignment when there are no match reasons

leonardost committed:
Fix change to style of "Free for the first year" in signup mobile view

leonardost commented:
Hey @poligilad-auto, I've fixed the layout observations you made, could you take another look?

> That's a setup in the right direction :) I'm not fan of this type of loading state, it's a bit outdated.

I feel you, I'm not a huge fan either 😄 I believe it's possible to skip the loading state and go straight to the next step, but I'm afraid it'd be more complicated than the previous updates. When we click the "Select" button, we do an availability check for the domain on the backend - if for whatever reason it isn't available (TLD on maintenance, already registered, etc.) a notice is shown for the user and they have to choose another one before proceeding.

<img width="1052" alt="Screen Shot 2021-08-09 at 18 53 48" src="https://user-images.githubusercontent.com/5324818/128779465-15af5589-daef-49f4-80f0-c4e63176bb45.png">

The shopping cart issues are somewhat similar, but related to the shopping cart code.

Changing those would involve moving or reworking some of the logic, which might get messy 😅 would it be ok if we postpone these improvements to a later iteration?

poligilad-auto commented:
Hey @leonardost,
Thanks for the changes and the explanation!

> Changing those would involve moving or reworking some of the logic, which might get messy 😅 would it be ok if we postpone these improvements to a later iteration?

Yes, sure. Let's postpone. It's not a big issue.

**All the changes are looking great, just 1 small thing to change and 1 thing I spotted on mobile:**
![image](https://user-images.githubusercontent.com/83961704/128832160-48721eca-814b-48ee-89e0-7a2e94c9ca7c.png)
1. Sale price: can you make the "/year" also `$font-body-small`
2. In mobile view can we make the domain name for the 2 suggested domains `$font-title-small`. They look to similar to the other results now.

That's it!

leonardost committed:
Remove duplicated imports

leonardost committed:
Increase size of "year" string in domain prices on sale

leonardost committed:
Increase size of featured domain names in mobile view

leonardost commented:
Hi @poligilad-auto, thanks for the review! I've fixed these two last observations now.

poligilad-auto reviewed (approved):
Nice job @leonardost 👌

leonardost committed:
Make all featured suggestions' buttons primary

leonardost committed:
Fix E2E tests

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6402588

Thank you @leonardost for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6402588).

tyxla commented on client/components/domains/domain-registration-suggestion/index.jsx:
```diff
@@ -238,15 +231,7 @@ class DomainRegistrationSuggestion extends React.Component {
-			comment: 'Shown next to a domain that has a special discounted sale price',
-		} );
-		const infoPopoverSize = isFeatured ? 22 : 18;
+		title = this.renderDomainParts( domain );
```
Nit: I've noticed that we always overwrite the previous value of `title` here, so the previous line is actually unnecessary. and we can directly make it a `const` here. Is this something you'd like to follow up with?

delputnam replied:
Oh! Good catch @tyxla. Thank you! I'll add a task for us (Nomado) to fix this.

tyxla replied:
Thanks a bunch, @delputnam! 🙌  No rush FWIW.
